### PR TITLE
Make type of error rate optional.

### PR DESCRIPTION
**Pull request by pkuyym**

### CODE REVIEW

File: deep_speech_2/evaluate.py [hunk 1]
```diff
@@ -111,6 +112,14 @@
...
     type=str,
     help="Vocabulary filepath. (default: %(default)s)")
+parser.add_argument(
+    "--error_rate_type",
+    default='wer',
+    choices=['wer', 'cer'],
+    type=str,
+    help="There are total two error rate types including wer and cer. wer "
+    "represents for word error rate while cer for character error rate. "
+    "(default: %(default)s)")
```
Comment (xinghai-sun) [inline deep_speech_2/evaluate.py]: Make the wording concise, e.g. `Error rate type for evaluation. 'wer' for word error rate and 'cer' for character error rate.`

File: deep_speech_2/evaluate.py [hunk 2]
```diff
@@ -153,10 +169,12 @@ def evaluate():
             for _, transcript in infer_data
         ]
         for target, result in zip(target_transcripts, result_transcripts):
-            wer_sum += wer(target, result)
+            error_sum += error_rate_func(target, result)
             num_ins += 1
-        print("WER (%d/?) = %f" % (num_ins, wer_sum / num_ins))
-    print("Final WER (%d/%d) = %f" % (num_ins, num_ins, wer_sum / num_ins))
+        print("%s (%d/?) = %f" % \
```
Comment (xinghai-sun) [inline deep_speech_2/evaluate.py]: Remove `\`

File: deep_speech_2/infer.py [hunk 3]
```diff
@@ -147,14 +156,22 @@ def infer():
         language_model_path=args.language_model_path,
         num_processes=args.num_processes_beam_search)
 
+    if args.error_rate_type == 'wer':
+        error_rate_func = wer
+        error_rate_info = 'wer'
+    else:
+        error_rate_func = cer
+        error_rate_info = 'cer'
+
```
Comment (xinghai-sun) [inline deep_speech_2/infer.py]: `error_rate_func = cer if args.error_rate_type == 'cer' else wer`

File: deep_speech_2/evaluate.py [hunk 4]
```diff
@@ -136,7 +145,14 @@ def evaluate():
         rnn_layer_size=args.rnn_layer_size,
         pretrained_model_path=args.model_filepath)
 
-    wer_sum, num_ins = 0.0, 0
+    if args.error_rate_type == 'wer':
```
Comment (xinghai-sun) [inline deep_speech_2/evaluate.py]: L148-L153 --> `error_rate_func = cer if args.error_rate_type == 'cer' else wer`

File: deep_speech_2/evaluate.py (hunk 2 above)
Comment (xinghai-sun) [inline deep_speech_2/evaluate.py]: Remove `error_rate_info` and use `print("Error rate [%s] ....." % args.error_rate_type)` for simplicity ?

**Review by xinghai-sun — CHANGES_REQUESTED** [review]

File: deep_speech_2/evaluate.py (hunk 1 above)
Comment (pkuyym) [reply deep_speech_2/evaluate.py]: Done.

File: deep_speech_2/evaluate.py (hunk 4 above)
Comment (pkuyym) [reply deep_speech_2/evaluate.py]: Done.

File: deep_speech_2/evaluate.py (hunk 2 above)
Comment (pkuyym) [reply deep_speech_2/evaluate.py]: Done.

Comment (pkuyym) [reply deep_speech_2/evaluate.py]: Done.

File: deep_speech_2/infer.py (hunk 3 above)
Comment (pkuyym) [reply deep_speech_2/infer.py]: Done.

**Review by kuke — COMMENTED** [review]
LGTM

**Review by xinghai-sun — APPROVED** [review]
LGTM.

**Pull request merged**